src/test/ui/array-slice-vec/infer_array_len.stderr

```diff
@@ -4,10 +4,10 @@ error[E0282]: type annotations needed
 LL |     let [_, _] = a.into();
    |         ^^^^^^
    |
-help: consider giving this pattern a type
+help: consider giving this pattern a type, where the placeholder `Type` is specified
    |
-LL |     let [_, _]: _ = a.into();
-   |               +++
+LL |     let [_, _]: Type = a.into();
```

Comment:
This isn't syntactical, but I think it may be more clear the user that it's their obligation to *do* something here if we print something like `: /* Type */ =`.

That's what we do for argument mismatches (see [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7db35c152de70325381878ff2b1db174)), and IMO it's a much more obvious way to point out that this is a placeholder that the users are obligated to fill in than either `: _` or `: Type`. 

I'm just worried about users going "wait, now it's telling me that `Type` isn't a valid struct or enum!" if they just try to apply the suggestion here, as they were doing when it was suggesting `: _`.